### PR TITLE
docs(ext): Update docs for URLPattern to make all examples work

### PR DESCRIPTION
**Pull request by kamilogorek**
Closes https://github.com/denoland/deno/issues/17845

### CODE REVIEW

File: ext/url/lib.deno_url.d.ts [hunk 1]
```diff
@@ -262,13 +262,14 @@ declare class URLPattern {
   /**
    * Test if the given input matches the stored pattern.
    *
-   * The input can either be provided as a url string (with an optional base),
-   * or as individual components in the form of an object.
+   * The input can either be provided as an absolute url string,
+   * relative url string with a base url, or as individual components
```
Comment (crowlKats) [inline ext/url/lib.deno_url.d.ts:266]: This is not completely accurate: a relative url with a base URL technically can be absolute as well. The previous wording was more generic

File: ext/url/lib.deno_url.d.ts [hunk 2]
```diff
@@ -248,10 +248,10 @@ declare interface URLPatternResult {
  * ```
  *
  * ```ts
- * // Specify a relative string pattern with a base URL.
- * const pattern = new URLPattern("/:article", "https://blog.example.com");
- * console.log(pattern.test("https://blog.example.com/article")); // true
- * console.log(pattern.test("https://blog.example.com/article/123")); // false
+ * // Specify a relative string pattern with a base url.
```
Comment (crowlKats) [inline ext/url/lib.deno_url.d.ts:251]:
```suggestion
 * // Specify a relative string pattern with a base URL.
```

**Review by crowlKats — CHANGES_REQUESTED** [review]

File: ext/url/lib.deno_url.d.ts (hunk 1 above)
Comment (kamilogorek) [reply ext/url/lib.deno_url.d.ts:266]: Agree, although the base is not optional if provided URL is relative. Not sure what should the wording be here tbh.

Comment (kamilogorek) [inline ext/url/lib.deno_url.d.ts:266]:
```suggestion
   * The input can either be provided as an absolute URL string with an optional base,
   * relative URL string with a required base, or as individual components
```

Maybe like that?

File: ext/url/lib.deno_url.d.ts [hunk 3]
```diff
@@ -283,13 +284,14 @@ declare class URLPattern {
   /**
    * Match the given input against the stored pattern.
    *
-   * The input can either be provided as a url string (with an optional base),
-   * or as individual components in the form of an object.
+   * The input can either be provided as an absolute url string,
+   * relative url string with a base url, or as individual components
```
Comment (kamilogorek) [inline ext/url/lib.deno_url.d.ts:288]:
```suggestion
   * The input can either be provided as an absolute URL string with an optional base,
   * relative URL string with a required base, or as individual components
```

Added the same suggestion here to not forget about updating it.

File: ext/url/lib.deno_url.d.ts (hunk 1 above)
Comment (crowlKats) [reply ext/url/lib.deno_url.d.ts:266]: yes that works

**Comment by bartlomieju** [comment]
@crowlKats please review and land this PR.

**Review by crowlKats — APPROVED** [review]
LGTM

**Pull request merged**